### PR TITLE
Bugfix/api key to access token

### PR DESCRIPTION
**Pull request by Pilopa**
## Proposed changes

Instead of rejecting access token generation for api keys, create an access token limited to the scope(s) of the api key.

### Your checklist for this pull request

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] PR is well described
- [x] Related issues linked to PR if existing and labels set
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [x] New code is covered with unit tests
- [x] CHANGELOG.md updated
- [ ] Definition of Done's fulfilled: [DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/definition-of-done.md) // [UI DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/ui-dod.md)
- [ ] Documentation updated (only necessary for new features or changed behaviour)

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 - Internal server error for git sub modules without tree object ([#1397](https://github.com/scm-manager/scm-manager/pull/1397))
 - Do not expose subversion commit with id 0 ([#1395](https://github.com/scm-manager/scm-manager/pull/1395))
+- Repositories can now be cloned with api keys ([#1407](https://github.com/scm-manager/scm-manager/pull/1407))
```
Comment (sdorra) [inline CHANGELOG.md:16]: I'm not sure if this is correct. I thought only mercurial repositories and clones over ssh with api keys are affected.

File: scm-webapp/src/main/java/sonia/scm/security/JwtAccessTokenBuilder.java [hunk 2]
```diff
@@ -156,7 +156,7 @@ private String getSubject(){
   @Override
   public JwtAccessToken build() {
     if (SecurityUtils.getSubject().getPrincipals().getRealmNames().contains(ApiKeyRealm.NAME)) {
-      throw new AuthorizationException("Cannot create access token for api keys");
+      scope = Scope.valueOf(SecurityUtils.getSubject().getPrincipals().oneByType(Scope.class));
     }
```
Comment (sdorra) [inline scm-webapp/src/main/java/sonia/scm/security/JwtAccessTokenBuilder.java:160]: Is this only relevant for api keys? What about a scoped JWT? Can i get more privileges, if i create a new JWT with a scoped JWT? Perhaps we should always use scope from the principals, if available?

@pfeuffer What do you think?

**Review by sdorra — CHANGES_REQUESTED** [review]

File: scm-webapp/src/main/java/sonia/scm/security/JwtAccessTokenBuilder.java (hunk 2 above)
Comment (pfeuffer) [reply scm-webapp/src/main/java/sonia/scm/security/JwtAccessTokenBuilder.java:160]: Sounds reasonable. If this is as simple as here, the `if` could be replaced with a check whether we have a `Scope` principal.

File: CHANGELOG.md (hunk 1 above)
Comment (Pilopa) [reply CHANGELOG.md]: We tested it and it does work over ssh with an api key. So it is only a problem with Mercurial.

**Pull request merged**